timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/v5/bdbdatafetchers/stores/BdbTripleStore.java

```diff
@@ -127,21 +129,29 @@ public void setPrefix(String cursor, String prefix, String iri) throws RdfProces
   @Override
   public void addRelation(String cursor, String subject, String predicate, String object, String graph)
       throws RdfProcessingFailedException {
-    new RelationDatabaseAction() {
 
-      @Override
-      public void executeAction(String key, String value) throws DatabaseException {
-        put(key, value);
-      }
-    }.execute(subject, predicate, object);
+    put(subject + "\n" +
+        predicate + "\n" + Direction.OUT.name(),
+        /*dataType*/ "\n" +
+        /*language*/ "\n" +
+        object
+    );
+    if (!predicate.equals(RDF_TYPE)) {
+      put(object + "\n" +
+          predicate + "\n" + Direction.IN.name(),
+            /*dataType*/ "\n" +
+            /*language*/ "\n" +
+          subject
+      );
+    }
   }
 
   @Override
   public void addValue(String cursor, String subject, String predicate, String value, String dataType, String graph)
       throws RdfProcessingFailedException {
     try {
       put(subject + "\n" +
-        predicate,
+        predicate + "\n" + Direction.OUT.name(),
```

Comment:
You're missing a comma. The key was `subject\npredicate` and is now `subject\npredicate\ndirection`